src/features/vote/hooks/usePriceRoundEvents.ts

```diff
@@ -0,0 +1,34 @@
+import provider from "common/utils/web3/createProvider";
+import createVotingContractInstance from "web3/createVotingContractInstance";
+import { useQuery } from "react-query";
+
+import {
+  queryPriceRoundEvents,
+  PriceRound,
+} from "web3/queryVotingContractEvents";
+
+const signer = provider.getSigner();
+const contract = createVotingContractInstance(signer);
+
+// This can be accessed without logging the user in.
+export default function usePriceRoundEvents() {
+  const { data, error, isFetching } = useQuery<PriceRound[]>(
+    "priceRoundEvents",
+    () => {
+      return queryPriceRoundEvents(contract).then((res) => {
+        if (res instanceof Error) return [];
+        if (res) {
+          return res;
+        } else {
+          return [];
+        }
+      });
+    }
+  );
+
+  if (data) {
+    return { priceRounds: data, error, isFetching };
+  } else {
+    return { priceRounds: [] as PriceRound[], error, isFetching };
```

Comment:
would be great if we didnt have to do this check before returning, and let the view check the the error /isFetching state to determine if it should use `priceRounds`.  The typescript would need to be changed to allow `undefined`. I'm also of the opinion that maybe we should just maintain the useQuery pattern and return any data as `data` and rename them in the view: `{data:priceRounds}`.   this would simplify a lot of these hooks:
```js

export default function usePriceRoundEvents() {
  return  useQuery<PriceRound[] | undefined>( "priceRoundEvents", () => {
      return queryPriceRoundEvents(contract)
    });
  );
}
```

 wdyt?

Reply:
Totally agree. I forgot to change this one over.